assets/src/components/detours/detourMap.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState } from "react"
+import { Shape } from "../../schedule"
+import { LatLngExpression } from "leaflet"
+import { Polyline, useMap, useMapEvent } from "react-leaflet"
+import Leaflet, { Map as LeafletMap } from "leaflet"
+import Map from "../map"
+import { CustomControl } from "../map/controls/customControl"
+import { Button } from "react-bootstrap"
+import { ReactMarker } from "../map/utilities/reactMarker"
+
+export const DetourMap = ({ shape }: { shape: Shape }) => {
+  const [startPoint, setStartPoint] = useState<LatLngExpression | null>(null)
+  const [endPoint, setEndPoint] = useState<LatLngExpression | null>(null)
+  const [detourPositions, setDetourPositions] = useState<LatLngExpression[]>([])
```

Comment:
random-thought(non-blocking): I think it's too early to refactor this now, but I think we should consider doing the following
1. Pull out this state into it's own hook, e.g., `useDetourCreationObject` or something.
2. Pass this state into `DetourMap` via props.
3. remove the existance of `startPoint` and `endPoint` and instead contain the implementation within a hook which provides an API.
    - I'm imagining something like a `state` variable, which would be a state machine. The states may be something like `awaitingStart` -> `awaitingPoints` -> `finished`, and then the `start` and `end` points are just the first and last elements of the `detourPositions`.
    - And a few functions, which can interact with the state machine, like `setStartPoint`, `addWayPoint`, `setEndPoint`.

Reply:
Yes, I really like this idea! Let's keep it in mind as we move forward and the interfaces between the different components and stuff start to settle down.